src/NServiceBus.Core/IConfiguredEndpoint.cs

```diff
@@ -0,0 +1,25 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using ObjectBuilder;
+
+    /// <summary>
+    /// Represents an endpoint in the start-up phase.
+    /// </summary>
+    public interface IConfiguredEndpoint
+    {
+        /// <summary>
+        /// Starts the endpoint and returns a reference to it.
+        /// </summary>
+        /// <param name="builder">The adapter for the containers resolve API.</param>
+        /// <returns>A reference to the endpoint.</returns>
+        Task<IEndpointInstance> Start(IBuilder builder);
+
+        /// <summary>
+        /// Allows lazy access to the message session (singleton) so that it can be registered in depenency injection.
+        /// Note: Only valid to use once the endpoint have started.
+        /// </summary>
+        Lazy<IMessageSession> MessageSession { get; }
```

Comment:
I don't understand how this is supposed to be used at all. Are we expecting users to just know what do with this?

Reply:
The main use case is asp.net core di integration. In this phase we provide that via a sample but the next step is to ship an integration package so users wouldn't really need to use this.

https://github.com/Particular/docs.particular.net/blob/docs-for-prepare-endpoint/samples/dependency-injection/aspnetcore/Core_7/Sample/ServiceCollectionExtensions.cs

We also have this documented https://github.com/Particular/docs.particular.net/blob/docs-for-prepare-endpoint/nservicebus/dependency-injection/index_nondefault_core_%5B7.2,).partial.md

Does this make sense?